lib/utils.js

```diff
@@ -319,3 +319,53 @@ export function promisifyCall(ctx, fn, ...args) {
     fn.apply(ctx, args);
   });
 }
+
+export function expandValues(valueList) {
```

Comment:
Comments on these utils would be nice when you clean everything up for the real PR